### PR TITLE
Fix: More robust connect cmesh faces by vertices and sphere cmesh examples

### PR DESCRIPTION
**Pull request by jmark**
**_Describe your changes here:_**

This PR introduces fixes/enhancements to make the `join_by_vertices` routine more robust regarding the value range of the input vertices. Besides that this PR makes the sphere cmesh examples more robust by applying the actual radius parameter at the end when the vertex information is stored in the cmesh object. For some use cases the radius gets extremely big, e.g. around 1e6, and the shell thickness, quite small, e.g. around 10. This might confuse the `join_by_vertices` routine. The fix is to determine the face connectivity on a reference sphere/shell with sane radius/thickness.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/DLR-AMR/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### License

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

**Comment by Davknapp** [comment]
> @Niklas997 @Davknapp
> 
> Do you need any input in order to progress on the review? Please agree on who feels responsible for doing the review.

I will do it

File: src/t8_cmesh/t8_cmesh_examples.cxx [hunk 1]
```diff
@@ -2932,9 +2940,8 @@ t8_cmesh_new_triangulated_spherical_surface_octahedron (const double radius, sc_
     all_eclasses[itree] = T8_ECLASS_TRIANGLE;
   }
 
-  double vertices_top[3][3] = { { radius, 0.0, 0.0 }, { 0.0, radius, 0.0 }, { 0.0, 0.0, radius } };
-
-  double vertices_bot[3][3] = { { radius, 0.0, 0.0 }, { 0.0, radius, 0.0 }, { 0.0, 0.0, -radius } };
+  double vertices_top[3][3] = { { 1.0, 0.0, 0.0 }, { 0.0, 1.0, 0.0 }, { 0.0, 0.0, 1.0 } };
+  double vertices_bot[3][3] = { { 1.0, 0.0, 0.0 }, { 0.0, 1.0, 0.0 }, { 0.0, 0.0, -1.0 } };
 
   int itree = -1;
```
Comment (Davknapp) [inline src/t8_cmesh/t8_cmesh_examples.cxx:2946]:
```suggestion
```

File: src/t8_cmesh/t8_cmesh_examples.cxx [hunk 2]
```diff
@@ -2932,9 +2940,8 @@ t8_cmesh_new_triangulated_spherical_surface_octahedron (const double radius, sc_
...
   for (int turn = 0; turn < ntrees / 2; turn++) {
```
Comment (Davknapp) [inline src/t8_cmesh/t8_cmesh_examples.cxx:2947]:
```suggestion
  for (int turn = 0, itree = -1; turn < ntrees / 2; turn++) {
```

File: src/t8_cmesh/t8_cmesh_examples.cxx [hunk 3]
```diff
@@ -3045,8 +3068,8 @@ t8_cmesh_new_triangulated_spherical_surface_icosahedron (const double radius, sc
   int itree = -1;
```
Comment (Davknapp) [inline src/t8_cmesh/t8_cmesh_examples.cxx:3068]:
```suggestion
```

File: src/t8_cmesh/t8_cmesh_examples.cxx [hunk 4]
```diff
@@ -3045,8 +3068,8 @@ t8_cmesh_new_triangulated_spherical_surface_icosahedron (const double radius, sc
...
   for (int turn = 0; turn < 5; turn++) {
```
Comment (Davknapp) [inline src/t8_cmesh/t8_cmesh_examples.cxx:3069]:
```suggestion
  for (int turn = 0, itree = -1; turn < 5; turn++) {
```

Comment (Davknapp) [reply src/t8_cmesh/t8_cmesh_examples.cxx:3069]: Please update these changes to the similar loops.

**Review by Davknapp — CHANGES_REQUESTED** [review]
Some small changes, then we can merge.

**Comment by jmark** [comment]
> Some small changes, then we can merge.

I like it! Done.

**Review by Davknapp — APPROVED** [review]

**Pull request merged**